src/main/java/mate/academy/bookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,83 @@
+package mate.academy.bookstore.service.impl;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.CartItemRequestDto;
+import mate.academy.bookstore.dto.ShoppingCartDto;
+import mate.academy.bookstore.dto.ShoppingCartRequestDto;
+import mate.academy.bookstore.exception.EntityNotFoundException;
+import mate.academy.bookstore.mapper.ShoppingCartMapper;
+import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.model.CartItem;
+import mate.academy.bookstore.model.CartItemKey;
+import mate.academy.bookstore.model.ShoppingCart;
+import mate.academy.bookstore.model.User;
+import mate.academy.bookstore.repository.CartItemRepository;
+import mate.academy.bookstore.repository.ShoppingCartRepository;
+import mate.academy.bookstore.service.BookService;
+import mate.academy.bookstore.service.LocaleService;
+import mate.academy.bookstore.service.ShoppingCartService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final LocaleService localeService;
+    private final BookService bookService;
+
+    @Override
+    public void createFor(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCart.setId(user.getId());
```

Comment:
can we not set userId as a PK by hands here? and use `@MapsId` in entity for that reason? also if you'll have any problems with that - check the 22.05 QnA on 14:00, Bohdan had a question about it there